### PR TITLE
[Merged by Bors] -  feat: SubtractionMonoid instance for LinearOrderedAddCommGroupWithTop

### PR DESCRIPTION
**Pull request by Command-Master**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

See #13724 for an earlier version

### CODE REVIEW

File: Mathlib/Algebra/Order/AddGroupWithTop.lean [hunk 1]
```diff
@@ -116,3 +116,82 @@ instance : LinearOrderedAddCommGroupWithTop (WithTop α) where
...
+    {a : α} (h : a ≠ ⊤) :
+    a + -a = 0 :=
+  LinearOrderedAddCommGroupWithTop.add_neg_cancel a h
+
+@[simp]
+lemma add_eq_top : a + b = ⊤ ↔ a = ⊤ ∨ b = ⊤ := by
+  constructor
+  · intro h
+    by_contra nh
+    rw [not_or] at nh
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/AddGroupWithTop.lean:136]:
```suggestion
    by_contra! nh
```
maybe?

**Review by YaelDillies — APPROVED** [review]
Thanks! 🚀

maintainer merge

**Comment by kim-em** [comment]
bors d+

File: Mathlib/Algebra/Order/AddGroupWithTop.lean (hunk 1 above)
Comment (Command-Master) [reply Mathlib/Algebra/Order/AddGroupWithTop.lean:136]: This file doesn't import `by_contra!`

**Pull request closed without merging**